### ISSUE TITLE
Pandoc MD item lists

### ISSUE THREAD

**Issue by npat-efault** [jrblevin/markdown-mode#13]
Pandoc MarkDown supports item, or definition, lists. They look like this:

```
This is the item title
:    This is the definition paragraph which
     may span several lines
```

If I put the cursor over the definition paragraph and press M-q (fill paragraph), emacs messes up everything. This makes it totally unworkable for texts with this kind of lists.

As a quick fix I added the ":" charater to the regexp that recognizes list items. 

```
(defconst markdown-regex-list
  "^\\([ \t]*\\)\\([0-9#]+\\.\\|[\\*\\+-:]\\)\\([ \t]+\\)"
  "Regular expression for matching list items.")
```

This allows me to enter the definition above as:

```
This is the item title

:    This is the definition paragraph which
     may span several lines
```

and have M-q work. This is valid Pandoc Markdown, but it would be nice if I could omit the blank line.

Thanks for making markdown-mode available!

**Comment by npat-efault** [comment]
Here is description of definition-lists in the Pandoc manual:

http://pandoc.org/README.html#definition-lists

**Comment by jrblevin** [comment]
Thanks for this patch.  I like that this allows filling of definitions more thoroughly than 84a49f6, and it also allows one to use the list editing commands to move items up and down, etc.  However, doing it this way makes ordered/unordered list items and definition list items semantically equivalent, which has the potential to break other things.  (This regular expression is used 15 times elsewhere for syntax highlighting, etc.)

I'd prefer to do it in a way that treats definition lists separately where appropriate (but also still leveraging existing code where possible, like the list editing commands).  I think it would be best to create a new regular expression for definition lists and use it directly in the fill prefix and elsewhere as needed.

**Comment by npat-efault** [comment]
Thanks for considering.

Obviously my patch was a quick and dirty solution. No doubt there are much better ways to support definition lists.

**Comment by jrblevin** [comment]
This should be fixed now in 12c1289.  Also font lock, list item reordering, etc. should work too.
